actgen/agents/directed_qnet.py

```diff
@@ -48,15 +50,15 @@ def directed_update(self, states, actions, delta_update, n_updates, baseline_qne
                 with torch.no_grad():
                     # update for normal DQN
                     if self.agent_type == 'dqn':
-                        original_q_values = self.forward(s.float())  # (1, n_actions)
-                        q_target = original_q_values.clone()  # (1, n_actions)
+                        original_q_values = self.forward(s.float().to(self.device))  # (1, n_actions)
+                        q_target = original_q_values.clone().to(self.device)  # (1, n_actions)
                         q_target[0][a] = float(original_q_values[0][a]) + delta_update  # (1, n_actions)
 
                         if self.pin_other_q_values:
                             get_current_q_vals = lambda s: self.forward(s.float())  # (1, n_actions)
                         else:
                             a = torch.as_tensor([a])  #(1)
-                            q_target = extract(q_target, a, idx_dim=-1)  # (1)
+                            q_target = extract(q_target, a, idx_dim=-1).to(self.device)  # (1)
```

Comment:
I wouldn't expect you to need an extra `to(device)` if all you're doing is operations on tensors that are already on the device you want.